### PR TITLE
rippled-validator-keys-tool: 20180927-d7774bc → 0.3.2

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Fix build that's been broken on hydra since 2019 (https://hydra.nixos.org/build/136814227). Prior failure was noted here: https://github.com/NixOS/nixpkgs/pull/112951#issuecomment-778687799

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
[mcgibbon@pn50:~/projects/nixpkgs]$  result/bin/validator-keys 
validator-keys [options] <command> [<argument> ...]
General Options:
  -h [ --help ]         Display this message.
  --keyfile arg         Specify the key file.
  -u [ --unittest ]     Perform unit tests.
  --version             Display the build version.

Commands: 
     create_keys                   Generate validator keys.
     create_token                  Generate validator token.
     revoke_keys                   Revoke validator keys.
     sign <data>                   Sign string with validator key.
     show_manifest [hex|base64]    Displays the last generated manifest
     set_domain <domain>           Associate a domain with the validator key.
     clear_domain                  Disassociate a domain from a validator key.
     attest_domain                 Produce the attestation string for a domain.

[mcgibbon@pn50:~/projects/nixpkgs]$  result/bin/validator-keys  --version
validator-keys version 0.3.2
```

### CODE REVIEW

File: pkgs/servers/rippled/validator-keys-tool.nix [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+  version = "0.3.2";
 
-  src = fetchgit {
-    url = "https://github.com/ripple/validator-keys-tool.git";
-    inherit rev;
-    sha256 = "1hcbwwa21n692qpbm0vqy5jvvnf4aias309610m4kwdsnzfw0902";
+  src = fetchFromGitHub {
+    owner = "ripple";
+    repo = "validator-keys-tool";
+    rev = "5d7efcfeda3bdf6f5dda78056004a7c326321e9b";
```
Comment (SuperSandro2000) [inline pkgs/servers/rippled/validator-keys-tool.nix:10]: Requested tags https://github.com/ripple/validator-keys-tool/issues/23

File: pkgs/servers/rippled/validator-keys-tool.nix [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
+    "-Dep_procs=1"
+  ];
 
   installPhase = ''
+    runHook preInstall
     install -D validator-keys $out/bin/validator-keys
+    runHook postInstall
   '';
 
   meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/servers/rippled/validator-keys-tool.nix:29]: Maybe you could add yourself as a maintainer?

File: pkgs/servers/rippled/validator-keys-tool.nix [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
+    repo = "validator-keys-tool";
+    rev = "5d7efcfeda3bdf6f5dda78056004a7c326321e9b";
+    sha256 = "1irm8asp6plk9xw3ksf4fqnim8h0vj3h96w638lx71pga1h4zvmy";
   };
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ openssl boost zlib rippled ];
+  buildInputs = [ openssl boost zlib icu rippled ];
 
   hardeningDisable = ["format"];
```
Comment (SuperSandro2000) [inline pkgs/servers/rippled/validator-keys-tool.nix:17]: Did you check if this is still necessary?

Comment (rmcgibbo) [reply pkgs/servers/rippled/validator-keys-tool.nix:17]: I did. It doesn't build without it.

File: pkgs/servers/rippled/validator-keys-tool.nix (hunk 2 above)
Comment (rmcgibbo) [reply pkgs/servers/rippled/validator-keys-tool.nix:29]: I suppose I must (though I have no idea what this package does and have no desire to learn about cryptocurrencies)...

Comment (SuperSandro2000) [reply pkgs/servers/rippled/validator-keys-tool.nix:29]: You don't need to if you don't want to.

Comment (rmcgibbo) [reply pkgs/servers/rippled/validator-keys-tool.nix:29]: No, it's fine. I'm just being dramatic.

**Comment by SuperSandro2000** [comment]
It takes 13GB (!) to build this thing! What the hell.

**Pull request merged**